docs/concepts.md

```diff
@@ -1,9 +1,20 @@
+| Concept   | Description  |
+|-----------|--------------|
+| Span | Atomic unit of telemetry data |
+| Trace | Contains spans, tree structure shows parent/child relationships |
+| Metric | Values calculated using telemetry data |
+| Log | No duration, timestamped, emitted by services/loggers |
```

Comment:
I find it a bit weird to start the page with a table, and no link to the relevant sections. Maybe we can either add a text or link to the sections below?

Reply:
Added an overview section.